plasmapy/particles/exceptions.py

```diff
@@ -1,41 +1,61 @@
 """
-Collection of `Exceptions` and `Warnings` for PlasmaPy particles.
+Collection of exceptions and warnings for `plasmapy.particles`.
 """
 __all__ = [
-    "AtomicError",
-    "AtomicWarning",
+    "ParticleError",
+    "ParticleTypeError",
+    "ParticleValueError",
+    "ParticleWarning",
     "ChargeError",
     "InvalidElementError",
     "InvalidIonError",
     "InvalidIsotopeError",
     "InvalidParticleError",
-    "MissingAtomicDataError",
-    "MissingAtomicDataWarning",
+    "MissingParticleDataError",
+    "MissingParticleDataWarning",
     "UnexpectedParticleError",
 ]
 
 from plasmapy.utils import PlasmaPyError, PlasmaPyWarning
 
 
-class AtomicError(PlasmaPyError):
-    """An exception for errors in the `~plasmapy.particles` subpackage."""
+class ParticleError(PlasmaPyError):
+    """Base exception for errors in the `~plasmapy.particles` subpackage."""
 
     pass
 
 
-class MissingAtomicDataError(AtomicError):
+class ParticleTypeError(ParticleError, TypeError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect type was provided.
+    """
+
+    pass
+
+
+class ParticleValueError(ParticleError, ValueError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect value was provided.
+    """
+
+    pass
```

Comment:
Similar question to `ParticleTypeError`.  Is this going to be for when a wrong particle value is given (e.g. a lepton when expecting an ion, an atom when expecting an ion, etc.) or when a wrong value is given for some functionality in `plasmapy.particles`.  Again, to me it should be used for the former case and the later case still raises `ValueError`.